torch/utils/data/dataloader.py

```diff
@@ -309,7 +319,8 @@ def _shutdown_workers(self):
             while not self.data_queue.empty():
                 self.data_queue.get()
             for _ in self.workers:
-                self.index_queue.put(None)
+                for q in self.index_queue:
```

Comment:
This is wrong. When `ind_worker_queue` is False, you need to put `num_worker` Nones.

Reply:
I think this is correct. The cycle by ``for _ in self.workers`` gurantee that ``index_queue`` get the ``num_worker`` Nones. I have changed ``index_queue`` type from Queue to list of Queues. Because of this, I put None to all ``q`` in ``self.index_queue``. In case when ``ind_worker_queue`` if False, the ``self.index_queeue`` will list with only one Queue insight.

Reply:
Yes, I am aware that you changed `index_queue` to a list of queues. My point was that if `ind_worker_queue=False`, the single queue in `index_queue` should receive `num_worker` `None`s, rather than just one `None`.

Your new change is also incorrect. The behavior should be, if `ind_worker_queue=False`, put  `num_worker` `None`s into the single queue. Otherwise, put a `None` in each queue.

By the way, could you change the name `index_queue` to `index_queue` to avoid confusion? Thanks!

Reply:
OK, thank you for attention. Fixed.